Added a warning if primary config is missing _self_ (#1756)
facebookresearch/hydra: website/docs/tutorials/structured_config/4_defaults.md

  ...
```

### A Note about composition order

> The default composition order in Hydra is that values defined in a config are merged into values introduced from configs in the Defaults List - or in other words - overriding them.

Reviewer: ```suggestion
Hydra's default behavior is to override values in the Defaults List with values subsequently defined in the same config.
```

Author: The values are not in the defaults list. I know you are trying to simplify but I want to be accurate.

Reviewer: how about
> Hydra's default behavior is to override values introduced by the Defaults List with values subsequently defined in the same config.

Author: too convoluted.

Reviewer 2: nit:

To someone that knows hydra only as userr, this seems more clear to me (but I 'm probably missing something)

>  The default behavior in Hydra is for values defined in a config to override values from configs in the Defaults List.

This behavior can be unintuitive when your primary config is a Structured Config, like in the example above.
For example, if the primary config is:
```python {6}
@dataclass
class Config:
    defaults: List[Any] = field(default_factory=lambda:  [
        "debug/activate",
        # If you do not specify _self_, it will be appended to the end of the defaults list by default.
        "_self_"
    ])

    debug: bool = False
```
And `debug/activate.yaml` is overriding the `debug` flag to `True`, the composition order would be such that debug ends up being `False`.  

> To get the desired behavior, explicitly specify `_self_` before `debug/activate.yaml`:

Reviewer: ```suggestion
To change the composition order, explicitly specify `_self_` before `debug/activate.yaml`:
```

Author: The sentence does not make a lot of sense.
There are many ways to change the composition order but the suggestion is very specific.

Reviewer: You are right.   I'm just not sure if we should say `desired behavior` here though.

Author: How about 

> To get the intuitively expected behavior

Reviewer: People starting from Hydra 1.1 will have a different expectation of the order. how about
> To have the defaults lists override the subsequent config value,

Author: > To get `debug/activate.yaml` to override this config, explicitly specify `_self_` before `debug/activate.yaml`:

  ...
```

#### A Note about composition order
When Hydra composes a config, by default - any values defined in the config are overriding the values coming from configs specified in the Defaults List.

> If you specify a value, Hydra 1.1 will warn that you need to specify `_self_` in the Defaults List to indicate the desired composition order (The above example does not specify a value so there is no warning).

Author: I have a new section for this, what do you think?

![image](https://user-images.githubusercontent.com/376455/128414140-4382aec2-d8d7-440c-a8b4-c60a5d5f8089.png)

  ...
```

### A Note about composition order
The default composition order in Hydra is that values defined in a config are merged into values introduced from configs in the Defaults List - or in other words - overriding them.

> This behavior can be unintuitive when your primary config is a Structured Config, like in the example above.

Reviewer 2: This is definitely quite unintuitive, though seeing the `_self_` makes it a bit clearer.

Author: I agree. Unfortunately this behavior makes a ton of sense in all other use cases and making the behavior special in this unique cases will likely add even more confusion.
